sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -288,6 +288,114 @@ case class ArrayContains(left: Expression, right: Expression)
   override def prettyName: String = "array_contains"
 }
 
+/**
+ * Checks if the two arrays contain at least one common element.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(a1, a2) - Returns true if a1 contains at least an element present also in a2.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(array(1, 2, 3), array(3, 4, 5));
+       true
+  """, since = "2.4.0")
+case class ArraysOverlap(left: Expression, right: Expression)
+  extends BinaryExpression with ImplicitCastInputTypes {
+
+  private lazy val elementType = inputTypes.head.asInstanceOf[ArrayType].elementType
+
+  override def dataType: DataType = BooleanType
+
+  override def inputTypes: Seq[AbstractDataType] = left.dataType match {
```

Comment:
`Seq(ArrayType, ArrayType)`?

Reply:
There are similar functions, such as `array_union`(#21061), `array_intersect`(#21102), `array_except`(#21103), and maybe `concat`(#20858) which is slightly different though, to handle two (or more) arrays with the same element type.
I think we should use the same way to specify and check input types.

I'd like to discuss the best way for it here or somewhere else.
cc @kiszk @mn-mikke Do you have any suggestions?
Also cc @gatorsmile @cloud-fan

Reply:
If common functions for a method, which accepts two array with the same type, are provided as trait, it would be fine.

Reply:
In that case, sounds like `concat` implementation is a good example.

Reply:
yea if implicit type cast is not allowed for these functions.

Reply:
Just a quick note, there is a [dedicated type coercion rule](https://github.com/apache/spark/pull/20858/files#diff-383a8cdd0a9c58cae68e0a79295520a3) for `concat` functions. So if there was the trait you described, we could remove the rule.

Reply:
@mn-mikke I am not sure, since it is quite a strange case, since it allows also string and byte. I am not sure we can do this with implicit type cast.

Reply:
@mgaido91 Sorry, I should have been more explicit. I've been referring to the below case that I added into `FunctionArgumentConversion` due to enabling type coercion of array types.
```
case c @ Concat(children) if children.forall(c => ArrayType.acceptsType(c.dataType)) &&
  !haveSameType(children) =>
  val types = children.map(_.dataType)
  findWiderCommonType(types) match {
    case Some(finalDataType) => Concat(children.map(Cast(_, finalDataType)))
    case None => c
  }
```

Reply:
implicit type cast is allowed in Presto. I am pushing here a proposal of trait, let me know what you think about it. Thanks.